tsl/src/compression/api.c

```diff
@@ -544,6 +544,18 @@ compress_chunk_impl(Oid hypertable_relid, Oid chunk_relid)
 		 */
 		ts_chunk_constraints_create(cxt.compress_ht, compress_ht_chunk);
 		ts_trigger_create_all_on_chunk(compress_ht_chunk);
+
+		/* Detect and emit warning if poor compression ratio is found */
+		float compression_ratio = ((float) before_size.total_size / after_size.total_size);
+		float POOR_COMPRESSION_THRESHOLD = 1.0;
+		elog(ts_guc_enable_compression_ratio_warnings &&
+					 compression_ratio < POOR_COMPRESSION_THRESHOLD ?
+				 WARNING :
+				 DEBUG1,
+			 "size before compression: " INT64_FORMAT
```

Comment:
This probably needs better wording, its too dry and doesn't even include name of the chunk.

Reply:
agree with ante here that this needs a better error message
`poor compression rate detected in chunk "xyz"`, the numbers could be moved into errdetail. Additionally you should probably use ereport here